src/scanpy/preprocessing/_simple.py

```diff
@@ -628,6 +628,21 @@ def normalize_per_cell(
 DT = TypeVar("DT")
 
 
+@njit
+def _create_regressor_categorical(
+    X: np.ndarray, cats: int, filters: np.ndarray
```

Comment:
https://github.com/scverse/scanpy/pull/3353#discussion_r1836827147

Reply:
I asked the name be changed because `cats` implies categories which on a pandas object is the values, not size.